doc/adrs/183_add_support_for_project_dependencies.adoc

```diff
@@ -0,0 +1,40 @@
+= [ADR-183] Add support for project dependencies
+
+
+== Context
+
+https://github.com/eclipse-sirius/sirius-web/blob/master/doc/iterations/2025.2/add_support_for_the_import_of_libraries.adoc[This shape] describes how project can import libraries, but does not define how the project-to-library dependency is stored.
+
+We need to capture this information to implement future library import/upgrade strategies.
+For example: we need to know whether a part of a project comes from a read-only library (to prevent users from editing it), or what are the libraries imported by a project that can be upgraded to a newer version.
+
+== Decision
+
+A new table `project_dependency` with the following columns should be created
+
+- `projectId`
+- `libraryId`
+- `readOnly`: a boolean indicating whether the imported library is read-only or not
+
+An `IEditingContextProcessor` implementation should be provided to copy imported read-only libraries into studio projects when loading them.
+This solution reuses the approach described in https://github.com/eclipse-sirius/sirius-web/blob/master/doc/adrs/124_add_support_for_libraries.adoc[ADR 124].
+Nothing is done to handle imported non read-only libraries into studio projects: the default behavior is to assume they are already in the editing context (e.g. they have been compied into it on import).
+
+Downstream applications need to implement their own `IEditingContextProcessor` to handle libraries they publish.
+
+The `StudioEditingContextPersistenceFilter` should prevent the persistence of imported read-only libraries in studio project.
```

Comment:
The imported library should probably have a dedicated schema to not be persisted at all. Something like `sirius:/library/` or something. It should not be done for one specific use case but instead be a feature.